lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -525,32 +540,8 @@ Ufe::SceneItem::Ptr UsdUndoAddNewMaterialCommand::insertedChild() const
 
 bool UsdUndoAddNewMaterialCommand::CompatiblePrim(const Ufe::SceneItem::Ptr& target)
 {
-    if (!target) {
-        return false;
-    }
-
-    // Must be a scope.
-    if (target->nodeType() != "Scope") {
-        return false;
-    }
-
-    // With the magic name.
-    if (target->nodeName() == UsdUndoAssignNewMaterialCommand::resolvedMaterialScopeName()) {
-        return true;
-    }
-
-    // Or with only materials inside
-    auto scopeHierarchy = Ufe::Hierarchy::hierarchy(target);
-    if (scopeHierarchy) {
-        for (auto&& child : scopeHierarchy->children()) {
-            if (child->nodeType() != "Material") {
-                // At least one non material
-                return false;
-            }
-        }
-    }
-
-    return true;
+    // Must be a materials scope.
+    return isMaterialsScope(target);
```

Comment:
Factored out into a new function, so that it can be reused by other commands in the future.